test/db/test_db.py

```diff
@@ -77,8 +78,31 @@
     delete_training_template,
     export_job_create,
 )
+from sqlalchemy.ext.asyncio import create_async_engine
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.ext.asyncio import AsyncSession
+
+# Create a unique test directory using absolute paths to prevent contamination
+TEST_BASE_DIR = os.path.abspath(os.path.join(tempfile.gettempdir(), f"transformerlab_db_test_{uuid.uuid4().hex[:8]}"))
+os.makedirs(TEST_BASE_DIR, exist_ok=True)
+
+# Set environment variables BEFORE any transformerlab imports
+os.environ["TFL_HOME_DIR"] = TEST_BASE_DIR
```

Comment:
You can't move these environment variables settings down here as they are used by the import for db. They have to be set before db is imported.